Libplanet/Blockchain/BlockChain.cs

```diff
@@ -1559,7 +1559,20 @@ bool buildStateReferences
                         )
                     );
 
-                StateStore.SetStates(blockHash, totalDelta, blockHash => this[blockHash]);
+                StateStore.SetStates(blockHash, totalDelta, blockHash =>
+                {
+                    if (ContainsBlock(blockHash))
+                    {
+                        return this[blockHash];
+                    }
+
+                    if (block.Hash.Equals(blockHash))
+                    {
+                        return block;
+                    }
+
+                    throw new KeyNotFoundException();
```

Comment:
How about this?
```suggestion
                    if (block.Hash.Equals(blockHash))
                    {
                        return block;
                    }
                    
                    return this[blockHash];
```

Reply:
It seems able to cover https://github.com/planetarium/libplanet/pull/968#discussion_r477921203. Thanks to suggestion!